static/js/settings_users.js

```diff
@@ -523,9 +523,17 @@ function handle_human_form(tbody, status_field) {
             return;
         }
 
+        let user_email = settings_data.email_for_user_settings(person);
+        if (!user_email) {
+            // This is necessary as email_for_user_settings returns undefined if
+            // email_address_visibiility is set to 'Nobody' and that will result in
+            // email field being empty in the form.
+            user_email = person.email;
+        }
```

Comment:
Hmm, should we just make `email_for_user_settings` return `person.email` for the nobody case?  It seems like that would be the cleaner interface.

Reply:
Hmm not sure, this is the only case where we show the custom email. Changing `email_for_user_settings` to return email when `show_email` is false, we would require the following changes -
```diff
diff --git a/static/js/people.js b/static/js/people.js
index b581c50b19..271b965bb8 100644
--- a/static/js/people.js
+++ b/static/js/people.js
@@ -1246,7 +1246,10 @@ function safe_lower(s) {
diff --git a/static/js/people.js b/static/js/people.js
index b581c50b19..271b965bb8 100644
--- a/static/js/people.js
+++ b/static/js/people.js
@@ -1246,7 +1246,10 @@ function safe_lower(s) {
 }
 
 export function matches_user_settings_search(person, value) {
-    const email = settings_data.email_for_user_settings(person);
+    let email;
+    if (settings_data.show_email()) {
+        email = settings_data.email_for_user_settings(person);
+    }
 
     return safe_lower(person.full_name).includes(value) || safe_lower(email).includes(value);
 }
diff --git a/static/js/settings_data.js b/static/js/settings_data.js
index 6d66f564ff..5cdd278f79 100644
--- a/static/js/settings_data.js
+++ b/static/js/settings_data.js
@@ -49,7 +49,7 @@ export function show_email() {
 
 export function email_for_user_settings(person) {
     if (!show_email()) {
-        return undefined;
+        return person.email;
     }
 
     if (person.delivery_email && user_can_access_delivery_email()) {
diff --git a/static/js/settings_users.js b/static/js/settings_users.js
index e51daf756a..2f5be59073 100644
--- a/static/js/settings_users.js
+++ b/static/js/settings_users.js
@@ -230,7 +230,9 @@ function human_info(person) {
     info.can_modify = page_params.is_admin;
     info.is_current_user = people.is_my_user_id(person.user_id);
     info.cannot_deactivate = info.is_current_user || (person.is_owner && !page_params.is_owner);
-    info.display_email = settings_data.email_for_user_settings(person);
+    if (settings_data.show_email()) {
+        info.display_email = settings_data.email_for_user_settings(person);
+    }
 
     if (info.is_active) {
         // TODO: We might just want to show this
diff --git a/static/js/typeahead_helper.js b/static/js/typeahead_helper.js
index e292ffc0aa..5796bf8a6d 100644
--- a/static/js/typeahead_helper.js
+++ b/static/js/typeahead_helper.js
@@ -89,7 +89,9 @@ export function render_person(person) {
         user_circle_class,
         is_person: true,
     };
-    typeahead_arguments.secondary = settings_data.email_for_user_settings(person);
+    if (settings_data.show_email()) {
+        typeahead_arguments.secondary = settings_data.email_for_user_settings(person);
+    }
     return render_typeahead_item(typeahead_arguments);
 }
```

Reply:
I feel the human_form case is exception where we show the email which is not deliver email, so I think keeping it as it is will be fine.

Reply:
I think it's possible the version you have there is better, or maybe we should have two functions (one that always returns a value; the other which returns a value if available)?  

But I'm not sure, and it's better to get this bug fix in than not, so I've merged this with some small comment updates.